src/Intervention/Image/Gd/Commands/CompareCommand.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Intervention\Image\Gd\Commands;
+
+use Intervention\Image\Comparison;
+use Intervention\Image\Gd\Color;
+use Intervention\Image\ImageManager;
+
+class CompareCommand extends \Intervention\Image\Commands\AbstractCommand
+{
+    /**
+     * Compare two images and produce a diff
+     *
+     * @param  \Intervention\Image\Image $image1
+     * @return boolean
+     */
+    public function execute($image)
+    {
+        $image = clone $image;
+        $diffImage = clone $image;
+        $otherImage = clone $this->argument(0)->required()->value();
+        $highlightColor = $this->argument(1)->value(new Color('rgb(255,0,0)'));
+        $resizeCanvas = $this->argument(2)->value(true);
+
+        $width = $image->width();
+        $height = $image->height();
+        $otherWidth = $otherImage->width();
+        $otherHeight = $otherImage->height();
+        $outputWidth = max($width, $otherWidth);
+        $outputHeight = max($height, $otherHeight);
+
+        if ($resizeCanvas) {
+            $image->resizeCanvas($outputWidth, $outputHeight, 'top-left');
+            $otherImage->resizeCanvas($outputWidth, $outputHeight, 'top-left');
+        } elseif ($width != $otherWidth or $height != $otherHeight) {
+            throw new \Intervention\Image\Exception\InvalidArgumentException(
+                'The two images are different sizes and you passed $resizeCanvas = false.'
+            );
+        }
+
+        $diffImageCore = $diffImage->getCore();
+        $imageCore = $image->getCore();
+        $otherImageCore = $otherImage->getCore();
+        $highlightColor = imagecolorallocate($diffImageCore, $highlightColor->r, $highlightColor->g, $highlightColor->b);
+
+        $diffPixelCount = 0;
+        for ($x = 0; $x < $width; $x++) {
+            for ($y = 0; $y < $height; $y++) {
+
+                $rgb1 = imagecolorat($imageCore, $x, $y);
+                $pix1 = imagecolorsforindex($imageCore, $rgb1);
+
+                $rgb2 = imagecolorat($otherImageCore, $x, $y);
+                $pix2 = imagecolorsforindex($otherImageCore, $rgb2);
+
+                if ($pix1 !== $pix2) {
```

Comment:
At this point we would need to come up with an algorithm saying something to the effect of. These colours are minimum 20% the same.